base/src/main/java/org/mozilla/jss/ssl/javax/JSSEngineReferenceImpl.java

```diff
@@ -1262,9 +1262,10 @@ public SSLEngineResult unwrap(ByteBuffer src, ByteBuffer[] dsts, int offset, int
             }
         } while (this_src_write != 0 || this_dst_write != 0);
 
-        if (seen_exception == false && ssl_exception == null) {
-            ssl_exception = checkSSLAlerts();
-            seen_exception = (ssl_exception != null);
+        SSLException checkException = checkSSLAlerts();
+        if (checkException != null && !seen_exception) {
+            ssl_exception = checkException;
+            seen_exception = ssl_exception != null;
         }
```

Comment:
I'm not really familiar with this code, but it looks like the original code would check for SSL alerts then stop checking once it has seen an exception, and the new code will always check for SSL alerts but discard all exceptions after the first one. Is that the intended change?

I think the new code can be simplified like this:
```
SSLException checkException = checkSSLAlerts();
if (checkException != null && !seen_exception) {
    ssl_exception = checkException;
    seen_exception = true;
}
```

Reply:
> I'm not really familiar with this code, but it looks like the original code would check for SSL alerts then stop checking once it has seen an exception, and the new code will always check for SSL alerts but discard all exceptions after the first one. Is that the intended change?

The problem with the previous code was that if an exception raise [here](https://github.com/dogtagpki/jss/blob/e22202201397a282a486a46454afc74ca6dc726d/base/src/main/java/org/mozilla/jss/ssl/javax/JSSEngineReferenceImpl.java#L1259) the method `checkSSLAlerts()` is never invoked and TLS alerts are not propagated (and audit are not written). With this change it is invoked in any case but the first error is reported.